recipes/leptonica/all/conanfile.py

```diff
@@ -84,10 +96,20 @@ def requirements(self):
         if self.options.with_webp:
             self.requires("libwebp/1.3.2")
 
+    def validate(self):
+        if Version(self.version) >= "1.84.0":
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C17, which your compiler does not support."
+                )
+
     def build_requirements(self):
         if self.options.with_webp or self.options.with_openjpeg:
             if not self.conf.get("tools.gnu:pkg_config", check_type=str):
                 self.tool_requires("pkgconf/2.1.0")
+        if Version(self.version) >= "1.84.0":
+            self.tool_requires("cmake/[>=3.21 <4]")
```

Comment:
```suggestion
        if Version(self.version) >= "1.84.0":
            self.tool_requires("cmake/[>=3.21 <4]")
```

The version 1.84.1 still uses CMake 3.1, or am I missing something?

https://github.com/DanBloomberg/leptonica/compare/1.83.1...1.84.1#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR11

Reply:
@uilianries
To enable C17, cmake/3.21 is required.